full-node/db/sov-schema-db/src/snapshot.rs

```diff
@@ -104,6 +115,129 @@ impl<Q: QueryManager> DbSnapshot<Q> {
             .expect("SchemaBatch lock must not be poisoned")
             .delete(key)
     }
+
+    /// Get value of largest key written value for given [`Schema`]
+    pub fn get_largest<S: Schema>(&self) -> anyhow::Result<Option<S::Value>> {
+        let local_cache = self
+            .cache
+            .lock()
+            .expect("SchemaBatch lock must not be poisoned");
+        let local_cache_iter = local_cache.iter::<S>();
+
+        let parent = self
+            .parents_manager
+            .read()
+            .expect("Parent lock must not be poisoned");
+
+        let parent_iter = parent.iter::<S>(self.id)?;
+
+        let mut combined_iter: SnapshotIter<'_, Q, S> = SnapshotIter {
+            local_cache_iter: local_cache_iter.peekable(),
+            parent_iter: parent_iter.peekable(),
+        };
+
+        if let Some((_, value)) = combined_iter.next() {
+            let value = S::Value::decode_value(&value)?;
+            return Ok(Some(value));
+        }
+
+        Ok(None)
+    }
+
+    /// Get value in [`Schema`] that is smaller or equal than give `seek_key`
+    pub fn get_prev<S: Schema>(
+        &self,
+        seek_key: &impl SeekKeyEncoder<S>,
+    ) -> anyhow::Result<Option<S::Value>> {
+        let seek_key = seek_key.encode_seek_key()?;
+
+        let local_cache = self
+            .cache
+            .lock()
+            .expect("Local cache lock must not be poisoned");
+
+        let local_cache_iter = local_cache.iter::<S>();
+
+        let parent = self
+            .parents_manager
+            .read()
+            .expect("Parent snapshots lock must not be poisoned");
+
+        let parent_iter = parent.iter::<S>(self.id)?;
+
+        let combined_iter: SnapshotIter<'_, Q, S> = SnapshotIter {
+            local_cache_iter: local_cache_iter.peekable(),
+            parent_iter: parent_iter.peekable(),
+        };
+
+        if let Some((_, value)) = combined_iter
```

Comment:
Here we keep `local_cache` & `parent` guards in scope so the method is locked during the filtering, is it intended?

Reply:
Yes. Because we need total ordering.

Reply:
ok